javascript/javascript2/week3/app.js

```diff
@@ -0,0 +1,126 @@
+document.addEventListener("DOMContentLoaded", function () {
+  setTimeout(logAfterTwoSeconds, 2500);
+});
+function logAfterTwoSeconds() {
+  console.log("Called after 2.5 seconds");
+}
+function logTextAfterSpecificTime(delay, stringToLog) {
+  const delayMs = delay * 1000;
+  setTimeout(function () {
+    console.log(stringToLog);
+  }, delayMs);
+}
+
+logTextAfterSpecificTime(5, "log this text after 5 seconds");
+
+button.addEventListener("click", function () {
+  logTextAfterSpecificTime(5, "called after 5 sec");
+});
+
+const saturnLogger = () => {
+  console.log("Saturn");
+};
+const earthLogger = () => {
+  console.log("Earth");
+};
+
+const planetLogFunction = (functionname) => {
+  functionname();
+};
+
+planetLogFunction(earthLogger);
+planetLogFunction(saturnLogger);
+
+buttonLogLocation.addEventListener("click", function () {
+  // get the current position
+  navigator.geolocation.getCurrentPosition(onSuccess);
+});
+
+function onSuccess(position) {
+  let { latitude, longitude } = position.coords;
+  console.log(latitude);
+  divCoordinates.textContent = `This is the latitude ${latitude}, this is the longitude ${longitude})`;
+}
+
+// 6
+let map;
+
+function initMap() {
+  const getCoords = () => {
+    navigator.geolocation.getCurrentPosition((pos) => {
+      latitude = pos.coords.latitude;
+      longitude = pos.coords.longitude;
+      map = new google.maps.Map(document.getElementById("map"), {
+        center: { lat: latitude, lng: longitude },
+        zoom: 8,
+      });
+      let marker = new google.maps.Marker({
+        position: { lat: latitude, lng: longitude },
+        map: map,
+      });
+    });
+  };
+
+  getCoords();
+}
+
+window.initMap = initMap;
+
+//7
+const runAfterDelay = (delay, callback) => {
+  setTimeout(callback, delay);
+};
+
+runAfterDelay(5000, function () {
+  console.log("logging text");
+});
+
+//8
+document.addEventListener("click", onClickCheck);
+
+let lastClicked;
+function onClickCheck() {
+  timeNow = new Date().getTime();
+  /*
+  testing in console
+  console.log(lastClicked);
+  console.log(timeNow);
+  console.log(timeNow - lastClicked);
+  */
+  if (timeNow < lastClicked + 500) {
+    // Execute the link action
+    console.log("double click!");
+  } else {
+    console.log("not double click!");
+  }
+
+  lastClicked = timeNow;
+}
+
+//9
+const jokeCreator = (shouldTellFunnyJoke, logFunnyJoke, logBadJoke) => {
+  if (typeof shouldTellFunnyJoke !== "boolean") {
+    throw new Error(`${shouldTellFunnyJoke} should be boolean`);
+  }
+  logFunnyJoke = () => {
+    console.log(
+      `And the bartender says, "Success, but you're not ready!" So a JavaScript function walks into a bar.`
+    );
+  };
+
+  logBadJoke = () => {
+    console.log(`Bad JavaScript joke`);
+  };
+
+  if (shouldTellFunnyJoke === true) {
+    logFunnyJoke();
+  } else if (shouldTellFunnyJoke === false) {
+    logBadJoke();
+  }
+};
+
+jokeCreator(true);
+jokeCreator(false);
```

Comment:
Great job! One thing I would suggest is to have shorter names for variables! Long names = more words = increase possibility of typos! But overall great job! 👏 👏

Reply:
Sorry but I think this is dangerous advice 😱 If you're worried about typos there are other tools to use like linters or TypeScript. 

Using longer names allows your code to be more expressive about your intent, code is read more often than it is written!

Reply:
Maybe I worded it wrong then but just wanted to clarify that I was referring to this specific code and I didn't mean to encourage anyone to be extreme and use a, or b as function names or so. My intent was to change the name of the functions like LogAfterTwoSeconds, which could easily be changed to delayedLog or at least change the seconds to sec. If the concern is with the readability of the code, there is always the option of adding extra comments. Also, it's not a good practice anyway to name a function that logs after 2.5 secs, LogAftertwoSeconds.(with this logic it should have been LogAfterTwoAndHalfSeconds). So my advice was to come up with shorter names that also contain enough information without overdoing it.